node/network/protocol/src/request_response/mod.rs

```diff
@@ -106,6 +120,8 @@ impl Protocol {
 			// assuming we can service requests relatively quickly, which would need to be measured
 			// as well.
 			Protocol::AvailabilityFetching => 100,
+			// 10 seems reasonable, considering group sizes of max 10 validators.
+			Protocol::CollationFetching => 10,
```

Comment:
continuing our discussion here https://github.com/paritytech/polkadot/pull/2594#discussion_r593902365

1. does substrate allow having multiple in-flight requests from the same peer? but even if not it would only take 10 nodes to prevent a collator from serving honest validators?
2. (unrelated here) what about the validator side: how do we ensure it has enough slots for collators and that they are not fully maliciously occupied? is it https://github.com/paritytech/polkadot/blob/6b474a4427accdcaca1d634d3cb7087b5b2ceec7/node/network/protocol/src/peer_set.rs#L59?

Reply:
ad 1) not really, only if the collator was really slow in processing requests, if it drains the queue fast enough - we won't have any problems. If not, then it is overwhelmed with the requests which could be the result of a DoS attack. In that case requests will be dropped. It is assumed that serving a collation is a rather quick process.
ad 2) yeah - we keep 25 incoming connections open, those are selected based on reputation. So an attacking node which does not deliver any value, will get disconnected over time.

Reply:
The issue is that we lose optionality. We may have a node occupying the peer-slot which is providing minimal value but we lose the option to connect a peer which may provide more value. But AFAIK req/res is not limited by the peer-set but rather the queue.